elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/TableContext.java

```diff
@@ -0,0 +1,268 @@
+    private final Map<String, Queryable> joins = new HashMap<>();
+
+    private final Handlebars handlebars = new Handlebars()
+                    .with(EscapingStrategy.NOOP)
+                    .registerHelper("sql", new Helper<Object>() {
+                        @Override
+                        public Object apply(final Object context, final Options options) throws IOException {
+                            return resolveSQLHandlebar(context, options);
+                        }
+                    });
+
+    public Object get(Object key) {
+
+        if (joins.containsKey(key)) {
+            Queryable joinQueryable = joins.get(key);
+            TableContext newCtx = TableContext.builder()
+                            .queryable(joinQueryable)
+                            .alias(appendAlias(alias, key.toString()))
+                            .metaDataStore(metaDataStore)
+                            .build();
+
+            joinQueryable.getColumnProjections().forEach(column -> {
+                newCtx.put(column.getName(), column.getExpression());
+            });
+
+            joinQueryable.getJoins().forEach((name, join) -> {
+                SQLTable joinTable = metaDataStore.getTable(join.getJoinTableType());
+                newCtx.addJoin(name, joinTable);
+            });
+
+            return newCtx;
+        }
+
+        // Physical References starts with $
+        if (key.toString().lastIndexOf('$') == 0) {
+            String resolvedExpr = alias + PERIOD + key.toString().substring(1);
+            return applyQuotes(resolvedExpr, queryable.getConnectionDetails().getDialect());
+        }
+
+        verifyKeyExists(key, super.keySet());
+
+        Object value = super.get(key);
+        if (key.toString().startsWith("$$")) {
+            return value;
+        }
+
+        return resolveHandlebars(this, key.toString(), value.toString(), emptyMap());
+    }
+
+    /**
+     * Resolves the handebars in column expression.
+     *
+     * @param tableCtx current {@link TableContext}.
+     * @param columnName column's name.
+     * @param columnExpr column's definition.
+     * @param callingColumnArgs If this is called from SQL helper then calling column's arguments.
+     * @return fully resolved column's expression.
+     */
+    @SuppressWarnings("unchecked")
+    public String resolveHandlebars(TableContext tableCtx, String columnName, String columnExpr,
+                    Map<String, Object> callingColumnArgs) {
+
+        Map<String, Object> tableArgs = new HashMap<>();
+        Map<String, Object> columnArgs = new HashMap<>();
+        Map<String, Object> requestTableArgs = new HashMap<>();
+        Map<String, Object> requestColumnArgs = new HashMap<>();
+
+        Queryable queryable = tableCtx.getQueryable();
+        Table table = metaDataStore.getTable(queryable.getName(), queryable.getVersion());
+
+        // Add the default argument values stored in metadata store.
+        if (table != null) {
```

Comment:
Do we ever expect table to be null?

Reply:
It can be null when it runs for Nested Queries.